system-addon/test/unit/unit-entry.js

```diff
@@ -1,6 +1,17 @@
 const {GlobalOverrider, FakePrefs, FakePerformance, EventEmitter} = require("test/unit/utils");
 const {chaiAssertions} = require("test/schemas/pings");
 
+// Cause React warnings to make tests that trigger them fail
+const origConsoleError = console.error; // eslint-disable-line no-console
+console.error = function(msg, ...args) { // eslint-disable-line no-console
+  if (/(Invalid prop|Failed prop type|Check the render method)/.test(msg)) {
+    throw new Error(msg);
+  }
+
+  // eslint-disable-next-line no-console
+  origConsoleError.apply(console, [msg, ...args]);
```

Comment:
nit: could just do this to avoid the triple disable line:
```js
Object.assign(console, { 
  _error: console.error, // eslint-disable-line no-console
  error(...args) {
    this._error(...args);
    if (/(Invalid prop|Failed prop type|Check the render method)/.test(args[0])) {
      throw new Error(args);
    }
  }
});
```

Also, should we always `console._error` and conditionally `throw`? Might be useful to see it "inline" where it would have appeared. Also, unclear if `msg` is sufficient or if there's interesting data in `args` (as you've originally defined) that would be lost if we only throw the first argument.

Reply:
To me, the trick to avoid the triple disable line makes the code unnecessarily hard to follow, so I think i'm going to leave that as is.  That said, I think you're totally right that we should always error and conditionally throw; I'll do that.